huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go

```diff
@@ -114,6 +246,67 @@ func resourceL7PolicyV3Create(ctx context.Context, d *schema.ResourceData, meta
 	return resourceL7PolicyV3Read(ctx, d, meta)
 }
 
+func buildRedirectPoolsExtendConfig(d *schema.ResourceData) *l7policies.RedirectPoolsExtendConfig {
+	var redirectPoolsExtendConfig *l7policies.RedirectPoolsExtendConfig
+	redirectPoolsExtendConfigRaw := d.Get("redirect_pools_extend_config").([]interface{})
+	log.Printf("[DEBUG] redirectPoolsExtendConfigRaw: %+v", redirectPoolsExtendConfigRaw)
+	if len(redirectPoolsExtendConfigRaw) == 1 {
+		redirectPoolsExtendConfig = &l7policies.RedirectPoolsExtendConfig{}
+		rewriteUrlConfigRaw := redirectPoolsExtendConfigRaw[0].(map[string]interface{})
+		redirectPoolsExtendConfig.RewriteUrlEnable = rewriteUrlConfigRaw["rewrite_url_enabled"].(bool)
+		redirectPoolsExtendConfig.RewriteUrlConfig = buildRewriteUrlConfig(rewriteUrlConfigRaw["rewrite_url_config"])
+	}
+	log.Printf("[DEBUG] redirectPoolsExtendConfig: %+v", redirectPoolsExtendConfig)
+	return redirectPoolsExtendConfig
+}
+
+func buildRewriteUrlConfig(data interface{}) *l7policies.RewriteUrlConfig {
+	var rewriteUrlConfig *l7policies.RewriteUrlConfig
+	rewriteUrlConfigRaw := data.([]interface{})
+	log.Printf("[DEBUG] rewriteUrlConfigRaw: %+v", rewriteUrlConfigRaw)
+	if len(rewriteUrlConfigRaw) == 1 {
+		if v, ok := rewriteUrlConfigRaw[0].(map[string]interface{}); ok {
+			rewriteUrlConfig = &l7policies.RewriteUrlConfig{}
+			rewriteUrlConfig.Host = v["host"].(string)
+			rewriteUrlConfig.Path = v["path"].(string)
+			rewriteUrlConfig.Query = v["query"].(string)
```

Comment:
```suggestion
			rewriteUrlConfig = &l7policies.RewriteUrlConfig{
				Host: v["host"].(string),
				Path: v["path"].(string),
				Query: v["query"].(string),
			}
```
please update them in other places

Reply:
done